packages/promotion/src/services/promotion-module.ts

```diff
@@ -809,12 +809,32 @@ export default class PromotionModuleService<
 
   @InjectTransactionManager("baseRepository_")
   async delete(
+    ids: string[] | string,
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<void> {
+    const idsToDelete = Array.isArray(ids) ? ids : [ids]
+
+    await this.promotionService_.delete(idsToDelete, sharedContext)
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async softDelete(
     ids: string | string[],
     @MedusaContext() sharedContext: Context = {}
   ): Promise<void> {
     const promotionIds = Array.isArray(ids) ? ids : [ids]
 
-    await this.promotionService_.delete(promotionIds, sharedContext)
+    await this.promotionService_.softDelete(promotionIds, sharedContext)
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async restore(
+    ids: string | string[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<void> {
+    const idsToRestore = Array.isArray(ids) ? ids : [ids]
+
+    await this.promotionService_.restore(idsToRestore, sharedContext)
```

Comment:
todo: align the soft remove and restore like we did in the sales and product module etc. Otherwise the link module will have issues